samples/nrf9160/ciphersuites/README.rst

```diff
@@ -0,0 +1,146 @@
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf9160dk_nrf9160_ns
+
+.. include:: /includes/spm.txt
+
+Overview
+********
+
+The sample first initializes the :ref:`nrfxlib:nrf_modem` and AT communications.
+Next, it provisions a root CA certificate to the modem using the :ref:`modem_key_mgmt` library.
+Provisioning must be done before connecting to the LTE network, because the certificates can only be provisioned when the device is not connected.
+
+The sample then iterates through a list of TLS ciphersuites, attempting connection to the host with each one of them.
+Ciphersuites that are supported by the host will connect successfully, while unsupported ciphersuites will cause a connection failure, setting ``errno`` to ``95``.
+
+Finally, the sample will end with a summary of the ciphersuites supported and unsupported by the host ``example.com``.
+
+Obtaining a certificate
+=======================
+
+The sample connects to ``www.example.com``, which requires an X.509 certificate.
+This certificate is provided in the :file:`samples/nrf9160/https_client/cert` folder.
+
+To connect to other servers, you might need to provision a different certificate.
+See :ref:`cert_dwload` for more information.
+
+Using Mbed TLS and TF-M
+***********************
+
+This sample supports using Mbed TLS and Trusted Firmware-M (TF-M).
+Instead of offloading the TLS sockets into the modem, you can use the Mbed TLS library from Zephyr.
+Using the Zephyr Mbed TLS, you can still use the offloaded sockets.
+Mbed TLS offers more configuration options than using the offloaded TLS handling.
+
+When using TF-M and Mbed TLS with PSA crypto, all the crypto operations are run on the secure side on the device.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/nrf9160/https_client`
+
+.. include:: /includes/build_and_run_nrf9160.txt
+
+To build the sample with Mbed TLS and TF-M, add the following to your west build command:
+
+.. code-block:: none
+
+   -DOVERLAY_CONFIG=overlay-tfm_mbedtls.conf
+
+Testing
+=======
+
+After programming the sample to your development kit, test it by performing the following steps:
+
+1. Connect the USB cable and power on or reset your nRF9160 DK.
+#. Open a terminal emulator and observe that the sample starts, provisions certificates and connects to the LTE network.
+#. Observe that the sample will iterate through a list of ciphersuites, attempting a connection to ``example.com`` with each one of them, showing either a successful or an unsuccessful connection.
+
+Sample output
+=============
+
+The sample shows the following output:
+
+.. code-block:: console
+
+   ciphersuites sample started
+   certificate match
+   waiting for network.. OK
+   trying all ciphersuites to find which ones are supported...
+   trying ciphersuite: TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
```

Comment:
I must say the errno code is quite confusing in such a scenario, it carries no information that there was a problem with the connection. I'm aware though that it's not something to be fixed in this PR.